### ISSUE TITLE
`clippy::doc_markdown` matches common database engine names (SQLite, MySQL, etc.)

### ISSUE THREAD

**Issue by LikeLakers2** [rust-lang/rust-clippy#16913]
### Summary

Pretty much the title. Clippy lints on `clippy::doc_markdown` when it detects common database engine names, such as:

* SQLite
* MySQL
* PostgresSQL
* MariaDB

### Lint Name

clippy::doc_markdown

### Reproducer

While writing a comment referencing SQLite:

```rust
#![warn(clippy::doc_markdown)]

/// Performs an SQLite call
fn _some_sqlite_call() {
    todo!()
}
```

`cargo clippy` returns the following warning:

```
warning: item in documentation is missing backticks
 --> src/lib.rs:3:17
  |
3 | /// Performs an SQLite call
  |                 ^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.95.0/index.html#doc_markdown
--snip--
help: try
  |
3 - /// Performs an SQLite call
3 + /// Performs an `SQLite` call
  |
```

This should not happen, as SQLite is referring to a piece of software - and from my understanding, clippy tends to allow software-related terms to bypass this lint.

### Version

```text
rustc 1.95.0 (59807616e 2026-04-14)
binary: rustc
commit-hash: 59807616e1fa2540724bfbac14d7976d7e4a3860
commit-date: 2026-04-14
host: x86_64-pc-windows-msvc
release: 1.95.0
LLVM version: 22.1.2
```

**Comment by QEEK-AI** [comment]
**QEEK.AI Agent :**

"Thanks for reporting this! You're absolutely right , database engine names like SQLite, MySQL, PostgreSQL, and MariaDB are proper nouns referring to specific software products, and they shouldn't trigger the `doc_markdown` lint".

The `doc_markdown` lint works by checking documentation for "items" that look like code (camelCase, snake_case, etc.) but aren't wrapped in backticks. It maintains an allowlist of words that are commonly used but shouldn't be linted (like "JavaScript", "GitHub", "iOS", etc.).

**The fix would be to add these database engine names to the exception list in `clippy_lints/src/doc.rs`. Here's roughly where that happens**:

```rust
// In clippy_lints/src/doc.rs
// The WORD_EXCEPTIONS constant contains words that should be allowed
const WORD_EXCEPTIONS: &[&str] = &[
    // ... existing exceptions
    "SQLite",
    "MySQL", 
    "PostgreSQL",
    "MariaDB",
    "MongoDB",
    "Redis",
    // etc.
];
```

**A few questions to help narrow this down:**
1. Should we include common variants like "Postgres" (without the SQL suffix) and "PG" (common shorthand for PostgreSQL)?
2. Are there other database engines that should be included? (e.g., "Oracle", "Cassandra", "CouchDB", "DynamoDB")

**If you'd like to contribute a fix, I'd be happy to review a PR that adds these to the exception list. Otherwise, this is a good first issue for anyone interested in contributing to Clippy!**
